tests/e2e/testsLibrary/che-code/ProjectAndFileTestsCheCode.ts

```diff
@@ -15,19 +15,68 @@ import { DriverHelper } from '../../utils/DriverHelper';
 import { CLASSES } from '../../inversify.types';
 import { Logger } from '../../utils/Logger';
 import { TimeoutConstants } from '../../TimeoutConstants';
+import { Workbench, SideBarView, ViewContent, ViewSection, ViewItem } from 'monaco-page-objects';
+import { TestConstants } from '../../TestConstants';
 
 @injectable()
 export class ProjectAndFileTestsCheCode {
 
+    private static readonly ENDPOINTS_TITLE_TEXT: string = 'endpoints';
+
     constructor(
         @inject(CLASSES.DriverHelper) private readonly driverHelper: DriverHelper) {}
 
-    public async waitWorkspaceReadinessForCheCodeEditor(): Promise<void> {
+    public async waitWorkspaceReadinessForCheCodeEditor(): Promise<Workbench> {
+        const polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING;
         try {
             await this.driverHelper.getDriver().wait(until.elementLocated(By.className('monaco-workbench')), TimeoutConstants.TS_SELENIUM_START_WORKSPACE_TIMEOUT);
+            const workbench: Workbench = new Workbench();
+            const sideBarView: SideBarView = workbench.getSideBar();
+            const sidebarViewContent: ViewContent = sideBarView.getContent();
+            let sidebarViewSections: ViewSection[];
+            while (true) {
+                sidebarViewSections = await sidebarViewContent.getSections();
+                Logger.debug(`ProjectAndFileTestsCheCode.waitWorkspaceReadinessForCheCodeEditor - Waiting for ${ProjectAndFileTestsCheCode.ENDPOINTS_TITLE_TEXT} sidebar section:`);
+                for (const section of sidebarViewSections) {
+                    Logger.trace(`Sidebar section: ${await section.getTitle()}`);
+                    if (await section.getTitle() === ProjectAndFileTestsCheCode.ENDPOINTS_TITLE_TEXT) {
+                        return workbench;
+                    }
+                }
+                await this.driverHelper.getDriver().sleep(polling);
+            }
         } catch (err) {
             Logger.error(`ProjectAndFileTestsCheCode.waitWorkspaceReadinessForCheCodeEditor - waiting for workspace readiness failed: ${err}`);
             throw err;
         }
     }
+
+    public async waitForRootProjectPresence(workbench: Workbench, sectionTitle: string, rootFolderName: string): Promise<ViewItem> {
```

Comment:
why cant we just check it like 

```
// it will throw error if will not find
projectSidebarViewSection = await new SideBarView().getContent().getSection(sectionTitle);
expect(await projectSidebarViewSection.findItem(rootFolderName)).not.eql(undefined);
```

Reply:
I'm trying to poll for the `ENDPOINTS` section. It's not present right after workspace start-up, takes some time for it to appear. I don't want to throw errors unless necessary.
I can however use your suggestion to simplify it a bit further down the logic tree though.

Reply:
hmm, I'm having issues using it for the `projectRootFolderElement`. I'll keep it as is for now.